### ISSUE TITLE
Building HIDAPI using CMake on Windows: Package 'libusb-1.0', required by 'virtual:world', not found

### ISSUE THREAD

**Issue by kuznetsov-m** [libusb/hidapi#408]
I trying to build HIDAPI DLL on Windows via cmake follow documentation.

### [Building HIDAPI from Source](https://github.com/libusb/hidapi/blob/master/BUILD.md#building-on-windows)
> Prerequisites:
> Windows:
> On Windows you just need a compiler. You may use Visual Studio or Cygwin/MinGW, depending on which environment is best for your needs.
> Building on Windows
> Any windows builds (MSVC or MinGW/Cygwin) are also supported by [CMake](https://github.com/libusb/hidapi/blob/master/BUILD.cmake.md).

Ok, I have CMake in OS.

### [Building HIDAPI using CMake](https://github.com/libusb/hidapi/blob/master/BUILD.cmake.md)
> Standalone package build
> cd `<build dir>`
> cmake `<HIDAPI source dir>`

I got a next log:
```
-- The C compiler identification is GNU 11.3.0
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /mingw64/bin/cc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- hidapi: v0.12.0
-- Found PkgConfig: /mingw64/bin/pkg-config.exe (found version "1.8.0")
-- Checking for module 'libusb-1.0>=1.0.9'
--   Package 'libusb-1.0', required by 'virtual:world', not found
CMake Error at /usr/share/cmake-3.22.1/Modules/FindPkgConfig.cmake:603 (message):
  A required package was not found
Call Stack (most recent call first):
  /usr/share/cmake-3.22.1/Modules/FindPkgConfig.cmake:825 (_pkg_check_modules_internal)
  libusb/CMakeLists.txt:15 (pkg_check_modules)
```

I downloaded and extracted libusb from https://libusb.info. But I don't understand how to use the library in CMake assembly.
**How to build HIDAPI DLL on Windows?**

**Comment by kuznetsov-m** [comment]
I looked at the [github workflows windows-cmake-mingw](https://github.com/libusb/hidapi/blob/master/.github/workflows/builds.yml#L142-L154) but I don't really see the libusb installation step there.
Then I don't understand what I'm doing wrong with running CMake.

**Comment by mcuee** [comment]
Which toolchain are you using? hidapi does not require libusb under Windows.

**Comment by kuznetsov-m** [comment]
@mcuee 
```
> cmake --version
cmake version 3.22.1
```

**Comment by mcuee** [comment]
Not CMake, but rather, are you using MSYS2 MinGW-w64 or other distro?

**Comment by kuznetsov-m** [comment]
Yes, I'm using MSYS2 MinGW-w64

**Comment by mcuee** [comment]
You should do this.

```
/c/work/libusb/hidapi_test/hidapi-hidapi-0.12.0-rc1/build_cmake_mingw64
$ cmake -G "MinGW Makefiles" ..
-- The C compiler identification is GNU 11.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/msys64/mingw64/bin/gcc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- hidapi: v0.12.0
-- Configuring done
-- Generating done
-- Build files have been written to: C:/work/libusb/hidapi_test/hidapi-hidapi-0.12.0-rc1/build_cmake_mingw64

 /c/work/libusb/hidapi_test/hidapi-hidapi-0.12.0-rc1/build_cmake_mingw64
$ mingw32-make
[ 50%] Building C object src/windows/CMakeFiles/hidapi_winapi.dir/hid.c.obj
[100%] Linking C shared library libhidapi.dll
[100%] Built target hidapi_winapi

```

**Comment by kuznetsov-m** [comment]
```
PS C:\projects\test_hidapi\build> cmake -G "MinGW Makefiles" ..\hidapi
CMake Error: Could not create named generator MinGW Makefiles

Generators
* Unix Makefiles               = Generates standard UNIX makefiles.
  Ninja                        = Generates build.ninja files.
  Ninja Multi-Config           = Generates build-<Config>.ninja files.
  CodeBlocks - Ninja           = Generates CodeBlocks project files.
  CodeBlocks - Unix Makefiles  = Generates CodeBlocks project files.
  CodeLite - Ninja             = Generates CodeLite project files.
  CodeLite - Unix Makefiles    = Generates CodeLite project files.
  Eclipse CDT4 - Ninja         = Generates Eclipse CDT 4.0 project files.
  Eclipse CDT4 - Unix Makefiles= Generates Eclipse CDT 4.0 project files.
  Kate - Ninja                 = Generates Kate project files.
  Kate - Unix Makefiles        = Generates Kate project files.
  Sublime Text 2 - Ninja       = Generates Sublime Text 2 project files.
  Sublime Text 2 - Unix Makefiles
                               = Generates Sublime Text 2 project files.

PS C:\projects\test_hidapi\build> cmake -G "Ninja" ..\hidapi
CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
See also "/c/projects/test_hidapi/build/CMakeFiles/CMakeOutput.log".
```

I already reinstalled the toolchain
`>pacman -S mingw-w64-x86_64-toolchain`
but MinGW Makefiles still didn't show up in the list of generators

**Comment by mcuee** [comment]
Did you install cmake as well with `pacman -S mingw-w64-x86_64-cmake'?

**Comment by mcuee** [comment]
You should also try to run the above command under the MSYS2 MinGW-w64 prompt, not just under the PowerShell.

**Comment by kuznetsov-m** [comment]
So I managed to integrate msys2 into terminal. Thanks!
However, I still get the message
```
user@31B4 MINGW64 /c/projects/test_hidapi/build
$ cmake "MinGW Makefiles" ../hidapi
CMake Error: CMake was unable to find a build program corresponding to "Ninja".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
See also "/c/projects/test_hidapi/build/CMakeFiles/CMakeOutput.log".
```

**Comment by kuznetsov-m** [comment]
So I managed to integrate msys2 into terminal. Thanks!
However, I still get the message `Package 'libusb-1.0', required by 'virtual:world', not found`

```
mihail@31B4 MINGW64 /c/projects/test_hidapi/build
$ cmake "MinGW Makefiles" ../hidapi
-- The C compiler identification is GNU 11.3.0
System is unknown to cmake, create:
Platform/MINGW64_NT-10.0-19043 to use this system, please post your config file on discourse.cmake.org so it can be added to cmake
-- Detecting C compiler ABI info
System is unknown to cmake, create:
Platform/MINGW64_NT-10.0-19043 to use this system, please post your config file on discourse.cmake.org so it can be added to cmake
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /mingw64/bin/cc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- hidapi: v0.12.0
-- Found PkgConfig: /mingw64/bin/pkg-config.exe (found version "1.8.0")
-- Checking for module 'libusb-1.0>=1.0.9'
--   Package 'libusb-1.0', required by 'virtual:world', not found
CMake Error at /usr/share/cmake-3.22.1/Modules/FindPkgConfig.cmake:603 (message):
  A required package was not found
Call Stack (most recent call first):
  /usr/share/cmake-3.22.1/Modules/FindPkgConfig.cmake:825 (_pkg_check_modules_internal)
  libusb/CMakeLists.txt:15 (pkg_check_modules)


-- Configuring incomplete, errors occurred!
See also "/c/projects/test_hidapi/build/CMakeFiles/CMakeOutput.log".
```

**Comment by Youw** [comment]
> System is unknown to cmake

That's a bummer.

---
Try: `-G"MSYS Makefiles"`

**Comment by kuznetsov-m** [comment]
Many thanks for the advice and help. I realized that the problem is in my development toolkit.
Apparently the problem is in the version of the installed CMake.
I just reinstalled MSYS2 and reinstalled the following packages:
```
pacman -S --needed base-devel mingw-w64-x86_64-toolchain
pacman -S mingw-w64-x86_64-cmake
```
[This post on the SO led me to the right idea](https://stackoverflow.com/questions/65146595/how-to-install-and-configure-cmake-in-msys2)
